Code/Tools/AssetProcessor/native/unittests/PlatformConfigurationUnitTests.cpp

```diff
@@ -89,144 +52,205 @@ void PlatformConfigurationTests::StartTest()
+        expectedFiles << m_assetRootPath.absoluteFilePath("rootfile3.txt");
+        expectedFiles << m_assetRootPath.absoluteFilePath("unrecognised.file"); // a file that should not be recognised
+        expectedFiles << m_assetRootPath.absoluteFilePath("unrecognised2.file"); // a file that should not be recognised
+        expectedFiles << m_assetRootPath.absoluteFilePath("subfolder1/test/test.format"); // a file that should be recognised
+        expectedFiles << m_assetRootPath.absoluteFilePath("test.format"); // a file that should NOT be recognised
 
-        UNIT_TEST_EXPECT_TRUE(config.GetScanFolderCount() == 11);
-        UNIT_TEST_EXPECT_FALSE(config.GetScanFolderAt(0).IsRoot());
-        UNIT_TEST_EXPECT_TRUE(config.GetScanFolderAt(8).IsRoot());
-        UNIT_TEST_EXPECT_FALSE(config.GetScanFolderAt(0).RecurseSubFolders());
-        UNIT_TEST_EXPECT_TRUE(config.GetScanFolderAt(1).RecurseSubFolders());
-        UNIT_TEST_EXPECT_TRUE(config.GetScanFolderAt(2).RecurseSubFolders());
-        UNIT_TEST_EXPECT_FALSE(config.GetScanFolderAt(8).RecurseSubFolders());
+        expectedFiles << m_assetRootPath.absoluteFilePath("GameNameButWithExtra/somefile.meo"); // a file that lives in a folder that must not be mistaken for the wrong scan folder
+        expectedFiles << m_assetRootPath.absoluteFilePath("GameName/otherfile.meo"); // a file that lives in a folder that must not be mistaken for the wrong scan folder
 
-        // --------------------------- RECOGNIZER TEST ------------------------
+        for (const QString& expect : expectedFiles)
+        {
+            EXPECT_TRUE(CreateDummyFile(expect));
+        }
+    }
 
-        RecognizerPointerContainer results;
-        UNIT_TEST_EXPECT_TRUE(config.GetMatchingRecognizers(tempPath.absoluteFilePath("subfolder1/rootfile1.txt"), results));
-        UNIT_TEST_EXPECT_TRUE(results.size() == 2);
-        UNIT_TEST_EXPECT_TRUE(results[0]);
-        UNIT_TEST_EXPECT_TRUE(results[1]);
-        UNIT_TEST_EXPECT_TRUE(results[0]->m_name != results[1]->m_name);
-        UNIT_TEST_EXPECT_TRUE(results[0]->m_name == "txt files" || results[1]->m_name == "txt files");
-        UNIT_TEST_EXPECT_TRUE(results[0]->m_name == "txt files 2" || results[1]->m_name == "txt files 2");
+    FileStatePassthrough m_fileStateCache;
+    QDir m_assetRootPath;
+    PlatformConfiguration m_config;
+};
 
-        results.clear();
-        UNIT_TEST_EXPECT_FALSE(config.GetMatchingRecognizers(tempPath.absoluteFilePath("test.format"), results));
-        UNIT_TEST_EXPECT_TRUE(config.GetMatchingRecognizers(tempPath.absoluteFilePath("subfolder1/test/test.format"), results));
-        UNIT_TEST_EXPECT_TRUE(results.size() == 1);
-        UNIT_TEST_EXPECT_TRUE(results[0]->m_name == "format files that live in a folder called test");
+TEST_F(PlatformConfigurationTests, TestPlatformsAndScanFolders_FeedPlatformConfiguration_Succeeds)
+{
+    EXPECT_EQ(m_config.GetEnabledPlatforms().size(), 2);
```

Comment:
nit: provide context to magic numbers or ideally pull them from data.

Reply:
I created a separate task to add context to the magic numbers. This should be something we can improve after the migration.